modAionImpl/src/org/aion/zero/impl/sync/handler/RequestBlocksHandler.java

```diff
@@ -0,0 +1,152 @@
+package org.aion.zero.impl.sync.handler;
+
+import java.util.Arrays;
+import java.util.List;
+import org.aion.p2p.Ctrl;
+import org.aion.p2p.Handler;
+import org.aion.p2p.IP2pMgr;
+import org.aion.p2p.V1Constants;
+import org.aion.p2p.Ver;
+import org.aion.util.conversions.Hex;
+import org.aion.zero.impl.core.IAionBlockchain;
+import org.aion.zero.impl.sync.Act;
+import org.aion.zero.impl.sync.msg.RequestBlocks;
+import org.aion.zero.impl.sync.msg.ResponseBlocks;
+import org.aion.zero.impl.types.AionBlock;
+import org.slf4j.Logger;
+
+/**
+ * Handler for block range requests from the network.
+ *
+ * @author Alexandra Roatis
+ */
+public final class RequestBlocksHandler extends Handler {
```

Comment:
This thread cannot cause any failure in the overall fast sync process. It's very likely that some peers will send invalid blocks. Those blocks are simply ignored, i.e. not added to the list for imports. That's why issues here are logged to debug not error.

Reply:
We don't have the logic to not send the request implemented (it can be added though). However, there is functionality for ignoring the message on the side of the peer receiving the request.

Reply:
I created an issue for further enhancing #857 of the communication.